x-pack/elastic-agent/pkg/agent/transpiler/vars.go

```diff
@@ -55,6 +57,18 @@ func (v *Vars) Replace(value string) (Node, error) {
 			}
 			set := false
 			for _, val := range vars {
+				for name, provider := range v.fetchContextProviders {
+					if varPrefixMatched(val.Value(), name) {
+						fetchProvider := provider.(composable.FetchContextProvider)
+						fval, _ := fetchProvider.Fetch(val.Value())
+						result += value[lastIndex:r[0]] + fval
+						set = true
```

Comment:
I think this should ensure that the value did exist before setting `set` to true. We would want the `ErrNoMatch` to be raised if that secret didn't exist. If the template is okay with it not matching it could provide a default to ensure that the `ErrNoMatch` doesn't occur.